lib/jxl/passes_state.cc

```diff
@@ -42,10 +42,12 @@ Status InitializePassesSharedState(const FrameHeader& frame_header,
   }
 
   shared->quant_dc = ImageB(frame_dim.xsize_blocks, frame_dim.ysize_blocks);
-  if (!(frame_header.flags & FrameHeader::kUseDcFrame) || encoder) {
-    shared->dc_storage =
-        Image3F(frame_dim.xsize_blocks, frame_dim.ysize_blocks);
-  } else {
+
+  shared->dc_storage = Image3F(frame_dim.xsize_blocks, frame_dim.ysize_blocks);
+  shared->dc = &shared->dc_storage;
+
+  bool use_dc = !!(frame_header.flags & FrameHeader::kUseDcFrame);
```

Comment:
maybe `use_dc_frame`?

Reply:
(which reminds me: shouldn't we allocate dc_storage only if not using a DC frame?)

Reply:
Done.

(Not unlikely)

Reply:
Can we restore that logic and see if the buffer overflow is still resolved?

Reply:
Seems to work.
Also reallocated `dc_storage` to 0x0 image, so that problem would manifest itself earlier next time.
PTAL